pkg/cache/application.go

```diff
@@ -363,6 +363,7 @@ func (app *Application) Schedule() bool {
 		app.scheduleTasks(func(t *Task) bool {
 			return t.placeholder
 		})
+		app.removeCompletedTasks()
```

Comment:
We need to remove it from here to let the state transitions run for the Task. If we removing it prematurely, it cannot reach Completed state. Another solution is to call `Application.removeTask()` from the task itself but this seems cleaner.